DIPs/DIP1xxx-YS.md

```diff
@@ -0,0 +1,146 @@
+# Named arguments lite
+
+| Field           | Value                                                           |
+|-----------------|-----------------------------------------------------------------|
+| DIP:            | 1xxx                                                            |
+| Review Count:   | 0                                                               |
+| Author:         | Yuxuan Shui (yshuiv7@gmail.com)                                 |
+| Implementation: | N/A                                                             |
+| Status:         | Draft                                                                |
+
+## Abstract
+
+Named arguments adds a way to annotate the arguments passed to functions. It makes the intention of passed arguments more clear, which should results in code been more readable. It also allows the order of arguments being changed on the callee side without breaking user code.
+
+### Reference
+
+Various solutions have been suggested in the NewsGroup,  [1](https://forum.dlang.org/post/khcalesvxwdaqnzaqotb@forum.dlang.org) and [2](https://forum.dlang.org/post/n8024o$dlj$1@digitalmars.com).
+
+Library only solutions have also been attempted multiple times, see [1](https://forum.dlang.org/post/awjuoemsnmxbfgzhgkgx@forum.dlang.org) and [2](https://github.com/CyberShadow/ae/blob/master/utils/meta/args.d).
+
+There are also a couple DIPs proposed or in development, see [DIP88](https://wiki.dlang.org/DIP88) and [rikkimax](https://github.com/rikkimax/DIPs/blob/named_args/DIPs/DIP1xxx-RC.md)
+
+## Contents
+* [Rationale](#rationale)
+* [Description](#description)
+* [Breaking Changes and Deprecations](#breaking-changes-and-deprecations)
+* [Reviews](#reviews)
+* [Appendix](#appendix-grammar-changes)
+
+## Rationale
+
+Named arguments is a common programming language feature found in various dynamic or compiled languages. There are various motivation behind that, such as to make code more readable, and to pass arguments in arbitrary order.
+
+Unlike named arguments in other languages and previous DIPs, this DIP mainly aim to address the readability issue. This is to increase the chance of adaptation, while still keeping one of the major niceties of this feature.
+
+Consider this example:
+
+```d
+// What are these arguments?
+DecimalNumber product = CalculateProduct(values, 7, false, null);
+```
+
+It is really difficult to decipher what each argument actually means. This is a real existing problem. As even Google's C++ coding style guide tries to provide a solution for that (they suggest using comments, we will discuss later).
+
+This DIP also has the added benefits of preventing silent breakage when the meanings of function parameters changed, but their type don't (assuming the meanings are reflected in the names). For example:
+
+```d
+// Previous API: void draw_rect(int x, int y, int width, int height);
+// New API:
+void draw_rect(int x0, int y0, int x1, int y1);
+```
+
```

Comment:
Details! It's not enough to provide an example and expect the reader to intuit its meaning. The difference between width/height and x1/y1 and why there is breakage here may not be immediately obvious to anyone unfamiliar with graphics programming. Mention that the change will cause a compiler error in user code.

This is also a good place to say something about how this proposal enables unordered arguments, but doesn't address them.